utilities/segmentation_utils/flowreader.py

```diff
@@ -195,3 +211,319 @@ def preprocess(self, generator_zip):
+                    self.output_size[1],
+                    self.num_classes,
+                )
+            )
+
+        # preprocess and assign images and masks to the batch
+        for i in range(n):
+            if column:
+                raw_masks = np.zeros(
+                    (self.mini_batch, self.output_size[0] * self.output_size[1], 1)
+                )
+            else:
+                raw_masks = np.zeros(
+                    (self.mini_batch, self.output_size[0], self.output_size[1])
+                )
+
+            for j in range(self.mini_batch):
+                image_index = i * self.mini_batch + j
+                image = Image.open(
+                    os.path.join(self.image_path, batch_image_filenames[image_index])
+                ).resize(self.image_size, Image.ANTIALIAS)
+
+                image = np.array(image)
+                image = image / 255
+
+                mask = Image.open(
+                    os.path.join(self.mask_path, batch_mask_filenames[image_index])
+                ).resize(self.output_reshape)
+
+                mask = np.array(mask)
+                # image = image[:, :, self.channel_mask]
+
+                if self.preprocessing_enabled:
+                    if self.preprocessing_seed is None:
+                        image_seed = np.random.randint(0, 100000)
+                    else:
+                        state = np.random.RandomState(self.preprocessing_seed)
+                        image_seed = state.randint(0, 100000)
+
+                    (
+                        image,
+                        mask,
+                    ) = ImagePreprocessor.augmentation_pipeline(
+                        image,
+                        mask=mask,
+                        input_size=self.image_size,
+                        output_size=self.output_size,
+                        output_reshape=self.output_reshape,
+                        seed=image_seed,
+                        #!both preprocessing queues are assigned by this time
+                        image_queue=self.preprocessing_queue_image,  # type: ignore
+                        mask_queue=self.preprocessing_queue_mask,  # type: ignore
+                    )
+                if column:
+                    mask = np.reshape(mask, self.output_size)
+
+                batch_images[i, j, :, :, :] = image
+                # NOTE: this provides the flexibility required to process both
+                # column and matrix vectors
+                raw_masks[j, ...] = mask
+
+            batch_masks[i, ...] = ImagePreprocessor.onehot_encode(
+                raw_masks, self.output_size, self.num_classes
+            )
+
+        # chaches the batch
+        self.image_batch_store = batch_images
+        self.mask_batch_store = batch_masks
+
+        # required to check when to read the next batch
+
+    def __len__(self) -> int:
+        return int(np.floor(len(self.image_filenames) / float(self.mini_batch)))
```

Comment:
changed to self.mini_batch from self.batch_size as the datasize calculation was incorrect